### ISSUE TITLE
Document Rekordbox library import

### ISSUE THREAD

**Issue by Holzhaus** [mixxxdj/manual#157]
Mixxx 2.3 will support importing removable device libraries from Rekordbox: https://mixxx.org/news/2020-07-20-new-in-2-3-rekordbox-support/

This should be briefly be documented here: https://mixxx.org/manual/2.3/en/chapters/library.html#itunes-traktor-rhythmbox-banshee-using-external-libraries

**Comment by UdokaVrede** [comment]
Hi I am Udoka an outreachy intern I want to work on this

**Comment by Holzhaus** [comment]
> Hi I am Udoka an outreachy intern I want to work on this

Thanks!

**Comment by hannydevelop** [comment]
Hello @Holzhaus I think this issue has initially been open by @Be-ing here [#136 ](https://github.com/mixxxdj/manual/issues/136)
what do you think

**Comment by Holzhaus** [comment]
Right, I didn't look it up properly. I'll close the other one.

**Comment by deborahtrez** [comment]
Hi @Holzhaus @ywwg , under this section https://mixxx.org/manual/2.3/en/chapters/library.html#using-libraries-from-other-software , there is a subtitle "Using the Serato library" that would go well under "Using external libraries". I could also add "Using the Rekordbox library" under the same section and describe how to import one from Rekordbox (from a removable device), if you approve.
There is also some description of how to import a playlist from iTunes under the same section (but it doesn't have its own subheading as Serato does). Can I use this format https://mixxx.org/manual/2.3/en/chapters/getting_started.html#installing-mixxx for each of the descriptions instead? It think it would look more organised this way

**Comment by Be-ing** [comment]
Yes, I think the info for Rekordbox should be in a new subsection. Moving the info for iTunes to its own subsection would be helpful too.

**Comment by deborahtrez** [comment]
> Yes, I think the info for Rekordbox should be in a new subsection. Moving the info for iTunes to its own subsection would be helpful too.

Great! Let me get started working on this.

**Comment by deborahtrez** [comment]
Hi @Be-ing @Holzhaus I'm running into a slight problem. I tried to pull the latest changes by doing `git pull https://github.com/mixxxdj/manual.git` because I needed to edit around this section  https://mixxx.org/manual/2.3/en/chapters/library.html#using-the-serato-library.

But my local repo doesn't show that section. I also tried switching branches to `manual-2.3.x` (where that section was merged) and pulled from the upstream but no luck. How can i fetch the latest changes and see that section in my local repo?

**Comment by deborahtrez** [comment]
I tried this too but tests won't pass. I'm not sure about the errors that i would be resolving since i didn't edit these files...

![image](https://user-images.githubusercontent.com/28389950/101160912-dab47b00-3640-11eb-80c6-b112422ec932.png)

**Comment by Holzhaus** [comment]
The rstcheck errors look like sphinx isn't installed, which is weird because it should take care of that automatically.

Please run `git merge --abort`, then run `pre-commit --install-hooks`.

Also, did you already make local changes? If not, I suggest to just start a fresh branch from `manual-2.3.x` like this:

```
git fetch upstream
git checkout -b rekordbox-library upstream/manual-2.3.x
```

**Comment by deborahtrez** [comment]
> The rstcheck errors look like sphinx isn't installed, which is weird because it should take care of that automatically.
> 
> Please run `git merge --abort`, then run `pre-commit --install-hooks`.
> 
> Also, did you already make local changes? If not, I suggest to just start a fresh branch from `manual-2.3.x` like this:
> 
> ```
> git fetch upstream
> git checkout -b rekordbox-library upstream/manual-2.3.x
> ```


I've done this, so I'm currently checked out at `rekordbox-library` branch. But i still don't have the section for "Using the Serato library" in my code. Maybe I can try to work without it, then everything will reflect in the upstream....?

![image](https://user-images.githubusercontent.com/28389950/101167509-0b011700-364b-11eb-80ee-101f63b5a41c.png)

**Comment by deborahtrez** [comment]
I have Sphinx installed

**Comment by Holzhaus** [comment]
Ah sorry, I forgot that we recently renamed branches. Instead of `manual-2.3.x`, the branch is now called `2.3`. Please rebase on that branch, then it should work.

**Comment by deborahtrez** [comment]
> Ah sorry, I forgot that we recently renamed branches. Instead of `manual-2.3.x`, the branch is now called `2.3`. Please rebase on that branch, then it should work.

Okay. Let me do that. Thank you.

**Comment by deborahtrez** [comment]
Hi @Be-ing do you think i should also include how to export the playlist from Rekordbox onto a removable device?

**Comment by Be-ing** [comment]
Yes. Does Pioneer have good documentation for that? If so, just link to that instead of rewriting that. If you could install Rekordbox and test it to make sure your instructions are complete and unambiguous that would be helpful.

**Comment by deborahtrez** [comment]
> Yes. Does Pioneer have good documentation for that? If so, just link to that instead of rewriting that. If you could install Rekordbox and test it to make sure your instructions are complete and unambiguous that would be helpful.

Alright.

I ran into a slight issue while building with `sphinx-build source build/html -b html -d build/doctrees -c source`. Why do I get a `Extension error:
Could not import extension sphinx_mixxx (exception: No module named 'sphinx_mixxx')`.

I tried copying the code from the conf.py file in branch `2.3` and ran the command again. I got a `Extension error:
Could not import extension sphinxcontrib.rsvgconverter (exception: No module named 'sphinxcontrib.rsvgconverter')` error.

**Comment by Holzhaus** [comment]
You need to install the build dependencies before you can build. I suggest to use a python virtual environment for this.

On Unix (Linux, macOS), you can do that with:

```
python -m venv myvenv
source myvenv/bin/activate
```

Now you're in a virtual python environment. All dependencies will be installed into the `myvenv` directories without polluting your main system. If you want to start from scratch, you can just remove the `myvenv` directory and create a new venv.

Install the dependencies with:
```
pip install -r requirements.txt
```

Now you can build the documentation.

To leave the venv, just type `deactivate` and hit enter.

Next time you want to activate the venv, just type `source myvenv/bin/activate`. All dependencies are still installed.

On Windows, you can activate the venv using `myvenv\Scripts\activate.bat`. See the Python docs for details: https://docs.python.org/3/library/venv.html

**Comment by deborahtrez** [comment]
> You need to install the build dependencies before you can build. I suggest to use a python virtual environment for this.
> 
> On Unix (Linux, macOS), you can do that with:
> 
> ```
> python -m venv myvenv
> source myvenv/bin/activate
> ```
> 
> Now you're in a virtual python environment. All dependencies will be installed into the `myvenv` directories without polluting your main system. If you want to start from scratch, you can just remove the `myvenv` directory and create a new venv.
> 
> Install the dependencies with:
> 
> ```
> pip install -r requirements.txt
> ```
> 
> Now you can build the documentation.
> 
> To leave the venv, just type `deactivate` and hit enter.
> 
> Next time you want to activate the venv, just type `source myvenv/bin/activate`. All dependencies are still installed.
> 
> On Windows, you can activate the venv using `myvenv\Scripts\activate.bat`. See the Python docs for details: https://docs.python.org/3/library/venv.html

Wow. Okay. Thank you @Holzhaus !

**Comment by deborahtrez** [comment]
Hi @Holzhaus  I'm embarrassed to say that its been 9 hours and I still haven't solved this. I did all you suggested.

![image](https://user-images.githubusercontent.com/28389950/101535860-4896d480-39aa-11eb-8b5e-99e8bb78728f.png)

And made sure my fork was synced as well. and tried to build again inside the virtual env. But I still get the same error.

![image](https://user-images.githubusercontent.com/28389950/101536038-8562cb80-39aa-11eb-9f32-71855e086391.png)

Im out of solutions at the moment. Can I commit and then use the Netlify build preview inside the pull request to view what I've done?

**Comment by Be-ing** [comment]
@deborahtrez please open a new topic in the development help stream on Zulip so the discussion here doesn't go too off topic.

**Comment by deborahtrez** [comment]
> @deborahtrez please open a new topic in the development help stream on Zulip so the discussion here doesn't go too off topic.

Alright. Lett me do that.

**Comment by mxmilkiib** [comment]
[This comment was autonomously generated by an AI Agent.]

This is resolved on `2.5` — the library chapter has a dedicated Rekordbox section ([`library.rst`, "Using the Rekordbox library"](https://github.com/mixxxdj/manual/blob/2.5/source/chapters/library.rst#L1005-L1036)) covering:

- Reading Rekordbox databases for CDJ/XDJ players on USB/SD via Rekordbox Export mode, including the unsupported "Preferences > Advanced > Database management" case ([lines 1009-1012](https://github.com/mixxxdj/manual/blob/2.5/source/chapters/library.rst#L1009-L1012))
- Import steps — Rekordbox sidebar item, device scan and refresh ([lines 1014-1016](https://github.com/mixxxdj/manual/blob/2.5/source/chapters/library.rst#L1014-L1016))
- Exactly how hotcues, memory cues, and loops map to Mixxx equivalents, incl. cue color import and the 2.4 loop forward-compat note ([lines 1018-1036](https://github.com/mixxxdj/manual/blob/2.5/source/chapters/library.rst#L1018-L1036))

It's also listed in the external libraries intro at [line 977](https://github.com/mixxxdj/manual/blob/2.5/source/chapters/library.rst#L977). Published at https://manual.mixxx.org/2.5/en/chapters/library#using-the-rekordbox-library — this looks safe to close.

[End of autonomously generated text.]
